src/pipeline_v3/tests/integration/test_search_integration.py

```
@@ -34,17 +34,36 @@ class TestSearchIntegration:
-        # Use the centralized IndexManager which already has unique collection names
-        keyword_index = KeywordIndex(config=test_config)
-        index_manager = IndexManager(config=test_config)
+        # Mock the OpenAI embedding model to avoid API calls
+        with patch("llama_index.embeddings.openai.OpenAIEmbedding") as mock_embedding_class:
+            # Create a mock that returns consistent embeddings
+            mock_embedding = MagicMock()
+
+            def get_text_embedding(text):
+                # Generate consistent fake embeddings based on text hash
+                import hashlib
+                hash_val = int(hashlib.md5(text.encode()).hexdigest(), 16)
```

Ruff (0.11.9)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)